pkg/security/tests/network_device_test.go

```diff
@@ -133,3 +138,157 @@ func TestNetDevice(t *testing.T) {
 		}
 	})
 }
+
+func TestTCFilters(t *testing.T) {
+	checkKernelCompatibility(t, "RHEL, SLES and Oracle kernels", func(kv *kernel.Version) bool {
+		// TODO: Oracle because we are missing offsets
+		return kv.IsRH7Kernel() || kv.IsOracleUEKKernel() || kv.IsSLESKernel()
+	})
+
+	// skip the test to avoid nested namespaces issues
+	if testEnvironment == DockerEnvironment {
+		t.Skip("skipping tc filters test in docker")
+	}
+
+	// dummy rule to force the activation of netdev-related probes
+	rule := &rules.RuleDefinition{
+		ID:         "test_rule",
+		Expression: `dns.question.type == A`,
+	}
+
+	test, err := newTestModule(t, nil, []*rules.RuleDefinition{rule}, testOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var testModuleCleanedUp bool
+	defer func() {
+		if !testModuleCleanedUp {
+			test.Close()
+		}
+	}()
+
+	syscallTester, err := loadSyscallTester(t, test, "syscall_tester")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sleepExecutable := which(t, "sleep")
+
+	var newNetNSSleep *exec.Cmd
+	defer func() {
+		if newNetNSSleep != nil && newNetNSSleep.Process != nil {
+			_ = newNetNSSleep.Process.Kill()
+		}
+	}()
+
+	t.Run("attach_detach_filters", func(t *testing.T) {
+		newNetNSSleep = exec.Command(syscallTester, "new_netns_exec", sleepExecutable, "600")
+		err := newNetNSSleep.Start()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		sleepProcPid := uint32(newNetNSSleep.Process.Pid)
+		if sleepProcPid == 0 {
+			t.Fatal("pid of the sleep command is zero")
+		}
+
+		netNs := utils.NetNSPathFromPid(sleepProcPid)
+		time.Sleep(3 * time.Second) // wait a bit for the tc probes to be attach to the new interface
+		ingressExists, egressExists, err := tcFiltersExist(netNs, "lo", "classifier_ingress", "classifier_egress")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !ingressExists {
+			t.Error("Ingress tc classifier does not exist")
+		}
+		if !egressExists {
+			t.Fatalf("Egress tc classifier does not exist")
+		}
+
+		lonelyTimeoutDuration := 1 * time.Second
+		// force a first NetworkNamespaceDrift check to set the timeout for the newly created interface,
+		// basically mimicking what resolver.flushNamespaces() is doing
+		resolver := test.probe.GetResolvers().NamespaceResolver
+		probesCount := test.probe.GetResolvers().TCResolver.FlushInactiveProbes(test.probe.Manager, resolver.IsLazyDeletionInterface)
+		resolver.PreventNetworkNamespaceDrift(probesCount, lonelyTimeoutDuration)
+
+		// wait for the netns resolver go routine to perform the second NetworkNamespaceDrift check that will trigger the eviction of the lonely interface probes
+		sleepDuration := netnsresolver.FlushNamespacesPeriod + lonelyTimeoutDuration + 1*time.Second
+		t.Logf("Waiting %s for the NetworkNamespaceDrift netns resolver timeout to fire", sleepDuration.String())
+		time.Sleep(sleepDuration)
```

Comment:
I wonder if we can convert this to a signal being sent to a channel instead of just sleeping cuz with how loaded the CI infra is, there's a chance that the work may not be done with this sleep and it may result in flakiness.

Reply:
Changed this in the latest pushed version.